### PR TITLE
Extracted PackageKey from PackageId, moved PackageId to metadata

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #141 
Extracted `PackageKey` from` PackageId`, moved `PackageId` class to metadata package and made it implement `NuspecField`. Also changed `Nuspec.id()` return type to `NuspecField`

### CODE REVIEW

File: src/test/java/com/artipie/nuget/PackageKeyTest.java [hunk 1]
```diff
@@ -10,42 +10,25 @@
...
-public class PackageIdTest {
-
-    @Test
-    void shouldPreserveOriginal() {
-        final String id = "Microsoft.Extensions.Logging";
-        MatcherAssert.assertThat(
-            new PackageId(id).original(),
-            Matchers.is(id)
-        );
-    }
```
Comment (genryxy) [inline src/test/java/com/artipie/nuget/PackageKeyTest.java:26]: @olenagerasimova prorbably move this and next removed test cases to tests for class `PackageId`

**Comment by olenagerasimova** [comment]
@genryxy thanks, added test and moved methods

**Review by genryxy — DISMISSED** [review]
@olenagerasimova thanks, looks good to me

**Comment by olenagerasimova** [comment]
@g4s8 please, take a look/merge

**Comment by olenagerasimova** [comment]
@g4s8 thanks, corrected

**Review by g4s8 — APPROVED** [review]
@olenagerasimova thanks

**Pull request merged**